nginx-controller/nginx/nevermerger.go

```diff
@@ -0,0 +1,43 @@
+package nginx
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+type NeverMerger struct {
```

Comment:
The fact that we have the Merger interface is great! We can easily replace one Merger with another one. However,  I'd prefer the hostname checking not to be a part of it, because  it's a different thing, though you can make it fit into the Merger interface.